src/qcodes/dataset/database_extract_runs.py

```diff
@@ -168,3 +175,218 @@ def _extract_single_dataset_into_db(
+    existing_run_id = get_runid_from_guid(target_conn, dataset.guid)
+    if existing_run_id is not None:
+        log.info(f"Dataset {run_id} (GUID: {dataset.guid}) already exists in target database")
+        return "already_exists"
+    
+    # Check if dataset is completed
+    if not dataset.completed:
+        log.warning(f"Dataset {run_id} is not completed, copying as-is")
+        return _copy_dataset_as_is(dataset, target_conn, target_exp_id)
+    
+    try:
+        # Check if dataset has already been exported to NetCDF
+        existing_netcdf_path = dataset.export_info.export_paths.get("nc")
+        
+        if existing_netcdf_path is not None:
+            # Check if the existing export path matches the desired export path
+            existing_path = Path(existing_netcdf_path)
+            
+            # If the file exists at the expected location, skip export
+            if existing_path.exists() and existing_path.parent == export_path:
+                log.info(f"Dataset {run_id} already exported to NetCDF at {existing_netcdf_path}")
+                netcdf_export_path = existing_netcdf_path
+            else:
+                # Export to the new location
+                log.info(f"Dataset {run_id} was exported to different location, re-exporting to {export_path}")
+                dataset.export("netcdf", path=export_path)
+                netcdf_export_path = dataset.export_info.export_paths.get("nc")
+        else:
+            # Try to export to NetCDF
+            log.info(f"Attempting to export dataset {run_id} to NetCDF")
+            dataset.export("netcdf", path=export_path)
+            netcdf_export_path = dataset.export_info.export_paths.get("nc")
+        
+        # Check if export was successful
+        if netcdf_export_path is None:
+            log.warning(f"Failed to export dataset {run_id} to NetCDF, copying as-is")
+            return _copy_dataset_as_is(dataset, target_conn, target_exp_id)
+            
+        log.info(f"Dataset {run_id} available as NetCDF at {netcdf_export_path}")
+        
+        # Create metadata-only version by copying dataset structure without raw data
+        log.info(f"Creating metadata-only version of dataset {run_id}")
+        
+        with atomic(target_conn) as target_conn_atomic:
+            # Add run metadata to runs table, preserving original GUID and captured_run_id
+            # Using _add_run_to_runs_table as it's the appropriate API for dataset extraction
+            _, _, target_table_name = _add_run_to_runs_table(
+                dataset, target_conn_atomic, target_exp_id
+            )
+            
+            # Note: We deliberately don't populate the results table to keep only metadata
+            log.info(f"Successfully created metadata-only version of dataset {run_id}")
+        
+        return "exported"
+        
+    except Exception as e:
+        log.warning(f"Failed to export dataset {run_id} to NetCDF: {e}, copying as-is")
```

Comment:
please find a better way to include the exception's message and full stack trace into the warning log, just {e} is not going to be helpful for the users and developers wh would be analysing log messages.